core-plugins/src/main/java/co/cask/hydrator/plugin/transform/ProjectionTransform.java

```diff
@@ -127,10 +134,11 @@ public void transform(StructuredRecord valueIn, Emitter<StructuredRecord> emitte
     StructuredRecord.Builder builder = StructuredRecord.builder(outputSchema);
     for (Schema.Field inputField : inputSchema.getFields()) {
       String inputFieldName = inputField.getName();
-      if (fieldsToDrop.contains(inputFieldName)) {
+      if (!fieldsToKeep.isEmpty() && (!fieldsToKeep.contains(inputFieldName)) {
```

Comment:
Hmm if fieldsToKeep is specified and a field is not in fieldsToKeep, fieldsToDrop wouldn't contain it, so it wouldn't be dropped?